GeneratorInterface/Core/interface/GeneratorSmearedProducer.h

```diff
@@ -0,0 +1,29 @@
+#ifndef GeneratorInterface_Core_GeneratorSmearedProducer_h
+#define GeneratorInterface_Core_GeneratorSmearedProducer_h
+
+#include "FWCore/Framework/interface/EDProducer.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+
+namespace edm {
+  class ParameterSet;
+  class ConfigurationDescriptions;
+  class Event;
+  class EventSetup;
+  class HepMCProduct;
+}
+
+class GeneratorSmearedProducer : public edm::EDProducer {
```

Comment:
Yes, this new producer, as well as the modified vertex smeared producer, should be thread aware (e.g. one, stream, or global), just as all other modules should be. That should happen in due course.